### ISSUE TITLE
Automate release tag and branch creation and review release notes in PR

### ISSUE THREAD

**Issue by adilGhaffarDev** [metal3-io/cluster-api-provider-metal3#1923]
Tasks to do:
* [x] Create script that creates release notes in CHANGELOG folder. We already have a script that does that, we need to modify it so it takes RELEASE_TAG as input and creates notes according to that, like this: RELEASE_TAG=v1.8.2 make release-notes
* [x] Create GitHub action that does the following:
  * [x] Create and push release branch if the tag is rc or new minor release
   * [x] Create and push tags
   * [x] Create a draft release from the notes PR that was merged in the first step. Check this: https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/workflows/release.yaml 
* [ ] Create a script that creates and push mariadb tag (this can probably go in mariadb repo)

How will this workflow help us?
- It will make it easier to review release notes.
- The release team don’t have to worry about creating and pushing tags and branches. There is less possibility of human error.

/kind feature

**Comment by tuminoid** [comment]
> Tasks to do:
> 
> * [ ]  Create script that creates release notes in CHANGELOG folder. We already have a script that does that, we need to modify it so it takes RELEASE_TAG as input and creates notes according to that, like this: RELEASE_TAG=v1.8.2 make release-notes

It already does it.

`RELEASE_TAG ?= $(shell git describe --abbrev=0 2>/dev/null)` does take env and only defaults to git describe if it wasn't there.

> * [ ]  Create GitHub action that does the following:
>   
>   * [ ]  Create and push release branch if the tag is rc or new minor release

Not sure if we need to make a branch RC time yet, we're really not doing the RC process in a way that would give any benefits from branching.

>   * [ ]  Create and push tags

Whats the trigger for the action going to be if not pushed tag?

>   * [ ]  Create a draft release from the notes PR that was merged in the first step. Check this: https://github.com/kubernetes-sigs/cluster-api/blob/main/.github/workflows/release.yaml

Release notes PR will have the release note content iterated, and the actual release note will just contain content from the PR?

> * [ ]  Create a script that creates and push mariadb tag (this can probably go in mariadb repo)

We should actually stop tagging mariadb like we do, and start doing releases for it when needed.

**Comment by adilGhaffarDev** [comment]
> `RELEASE_TAG ?= $(shell git describe --abbrev=0 2>/dev/null)` does take env and only defaults to git describe if it wasn't there.

Yes, that part is already there. Whoever works on it make sure the name of the file is the same as the release and it is in the CHANGELOG folder. The only thing the release team has to do is run make target and open a PR with the newly added release notes file.

**Comment by adilGhaffarDev** [comment]
> Not sure if we need to make a branch RC time yet, we're really not doing the RC process in a way that would give any benefits from branching.

I agree that we need to document and decide when to create beta and when to create RC, but I would say if we create RC, we should create a branch too, otherwise we should just keep creating beta releases. 

Regarding the benefit of RC, I think it is better to have CI ready and running on the new branch earlier than the GA release. Lets document it when to do which release so its more clear to the release team.

**Comment by adilGhaffarDev** [comment]
> Release notes PR will have the release note content iterated, and the actual release note will just contain content from the PR?

Yes, the actual release notes will be the same as the PR.  
PR notes will reviewed and approved by the maintainers.

**Comment by adilGhaffarDev** [comment]
> We should actually stop tagging mariadb like we do, and start doing releases for it when needed.

I agree, lets start doing that too from next cycle.

**Comment by tuminoid** [comment]
> > We should actually stop tagging mariadb like we do, and start doing releases for it when needed.
> 
> I agree, lets start doing that too from next cycle.

Doing proper releases need to be agreed upon, as it means branching and test combination changes etc. But let's plan that we have it agreed in 4 months when next releases are coming out.

> > `RELEASE_TAG ?= $(shell git describe --abbrev=0 2>/dev/null)` does take env and only defaults to git describe if it wasn't there.
> 
> Yes, that part is already there. Whoever works on it make sure the name of the file is the same as the release and it is in the CHANGELOG folder. The only thing the release team has to do is run make target and open a PR with the newly added release notes file.

Just noticed thatt this is not working great right now btw, it compares wrong releases and often misses to produce proper output even. So there is much to improve here as well.

**Comment by Rozzii** [comment]
@adilGhaffarDev will check but this is most likely done

**Comment by adilGhaffarDev** [comment]
The only thing remaining for this is mariadb tag pushing. We have decided not to push anymore.
So we can close this issue.
